### DISCUSSION TITLE
Minimize config for specific module

### DISCUSSION THREAD

**Discussion by garudaonekh** [jgromes/RadioLib#638] [General]
Hi,
I want to LLCC68 for Atmega16, which has only 16KB space. It's too small for RadioLib.

Anyway to configure it for minimal size for specific LORA module only(LLCC68)?

Thanks;

**Comment by jgromes** [comment]
I suppose this depends on what do you want to do with RadioLib. "Only" LLCC68 module means that you also need the `SX1268`, `SX126x`, `Module` and `PhysicalLayer` classes due to inheritances. Sou you could try removing functionality you don't need - and hope that you can do better job of it than the compiler. You could also try to optimize the compiler for size (if it's not already).

This is not really an issue (as it doesn't need an action from RadioLib), so I'm converting to discussion instead.

**Reply by garudaonekh** [reply]
Yes, I have tried this before posting this issue but failed due to complicated hierarchy and inheritance. My goal is to reduce 40% of this library. Currently, it's
`Sketch uses 18120 bytes (114%) of program storage space. Maximum is 15872 bytes.`

If not possible, I will use Atmega32 instead.

**Reply by jgromes** [reply]
I don't think that's possible, you could try using some simpler library for SX126x. Or, if you really need the features in RadioLib, you might have to use a larger MCU. Small footprint was not one of the design goals, sorry.

**Reply by garudaonekh** [reply]
Thanks, I will use Atmega32 instead.
